coral/src/app/features/approvals/acls/components/AclApprovalsTable.tsx

```diff
@@ -0,0 +1,311 @@
+        status: "neutral",
+        text: environmentName,
+      }),
+    },
+    {
+      type: "text",
+      field: "teamname",
+      headerName: "Team",
+    },
+    {
+      type: "status",
+      field: "aclType",
+      headerName: "ACL type",
+      status: ({ aclType }) => ({
+        status: aclType === "CONSUMER" ? "success" : "info",
+        text: aclType,
+      }),
+    },
+    {
+      type: "status",
+      field: "requestStatus",
+      headerName: "Status",
+      status: ({ requestStatus }) => {
+        if (requestStatus === "") {
+          return {
+            status: "neutral",
+            text: "-",
+          };
+        }
+        return {
+          status: requestStatusChipStatusMap[requestStatus],
+          text: requestStatusNameMap[requestStatus],
+        };
+      },
+    },
+    {
+      type: "status",
+      field: "requestOperationType",
+      headerName: "Request type",
+      status: ({ requestOperationType }) => {
+        if (requestOperationType === "") {
+          return {
+            status: "neutral",
+            text: "-",
+          };
+        }
+        return {
+          status: requestOperationTypeChipStatusMap[requestOperationType],
+          text: requestOperationTypeNameMap[requestOperationType],
+        };
+      },
+    },
+    {
+      type: "text",
+      field: "username",
+      headerName: "Requested by",
+    },
+    {
+      type: "text",
+      field: "requesttimestring",
+      headerName: "Requested on",
+      formatter: (value) => {
+        return `${value}${"\u00A0"}UTC`;
+      },
+    },
+    {
+      headerName: "Details",
+      headerInvisible: true,
+      type: "custom",
+      UNSAFE_render: ({ id, topicname }: AclRequestTableRow) => {
+        return (
+          <GhostButton
+            icon={infoSign}
+            aria-label={`View acl request for ${topicname}`}
+            onClick={() => onDetails(String(id))}
+            dense
+          >
+            <span aria-hidden={"true"}>View details</span>
+            <span className={"visually-hidden"}>
+              View topic request for {topicname}
+            </span>
+          </GhostButton>
+        );
+      },
+    },
+    {
+      width: 30,
+      headerName: "Approve",
+      headerInvisible: true,
+      type: "custom",
+      UNSAFE_render: ({ id, requestStatus, topicname }: AclRequestTableRow) => {
+        const [isLoading, setIsLoading] = useState(false);
+        if (requestStatus === "CREATED") {
+          const approveIsLoading = isBeingApproved(String(id));
+          const declineIsLoading = isBeingDeclined(String(id));
+          return (
+            <GhostButton
+              onClick={() => {
+                setIsLoading(true);
+                onApprove(String(id));
+              }}
+              title={"Approve acl request"}
+              aria-label={`Approve acl request for ${topicname}`}
+              disabled={approveIsLoading || declineIsLoading}
```

Comment:
I think this disabled state will now not render as we expect it to. It was implemented with this PR: https://github.com/aiven/klaw/pull/732

The objective was to prevent users from firing several approve/decline requests at the same time, to not test the limits of the API and the UI in its current state.

- Click approve for req 23
- Req 23's approve button shows loadingIcon
- All other req buttons are disabled

So I believe that in this `disabled` prop, we should have the `isLoading` returned value from the approve/reject `useMutations`, not the returned values from `isBeingApproved` / `isBeingDeclined`

Reply:
> The objective was to prevent users from firing several approve/decline requests at the same time, to not test the limits of the API and the UI in its current state.


I see, thanks. I'll maybe add some global prop to the table that disables all actions if true. (in this case if either of the mutations are loading)

Reply:
That makes sense, thank you!